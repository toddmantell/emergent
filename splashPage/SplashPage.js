import React from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';
import SplashForm from './SplashForm';
import {routes} from '../constants';

export default class SplashPage extends React.Component {
  constructor() {
    super();

    this.redirectToStateList = this.redirectToStateList.bind(this);
  }
  static navigationOptions = {
    title: "Welcome"
  };
  redirectToStateList() {
    const { navigate } = this.props.navigation;

    navigate(routes.StateList);
  }
  render() {
    return (
      <SplashForm styles={styles} redirectToStateList={this.redirectToStateList} />
    );
  }
} 

const styles = StyleSheet.create({
  container: {
    flex: 2,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  button: {
    backgroundColor: 'midnightblue',
    flex: 2
  }
});
