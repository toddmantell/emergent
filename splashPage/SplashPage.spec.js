import React from 'react';
import SplashPage from './SplashPage';
import renderer from 'react-test-renderer';

const stubFunctionAsProp = () => 'stub!';

test("Splash Page Renders", () => {
    const mountedSplashPage = renderer.create(<SplashPage redirectToFindTherapist={stubFunctionAsProp} 
        style={{padding: 0}}/>).toJSON();

    const button = mountedSplashPage.children[3].children[0].children[0];
    console.log(button);

    expect(mountedSplashPage.children.length).toBe(4);
    expect(mountedSplashPage.children[3].type).toBe('View');
});