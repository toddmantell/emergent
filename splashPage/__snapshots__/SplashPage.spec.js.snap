// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 2,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "midnightblue",
      }
    }
  >
    Open up App.js to start working on your app!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "midnightblue",
      }
    }
  >
    Changes you make will automatically reload.
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "midnightblue",
      }
    }
  >
    Shake your phone to open the developer menu.
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "midnightblue",
            },
          ]
        }
      >
        Find A Therapist
      </Text>
    </View>
  </View>
</View>
`;
